File: scripts/test_solutions/150_evaluate_reverse_polish.js
var evalRPN = function(tokens) {
    const stack = [];
    for (const t of tokens) {
        if ('+-*/'.includes(t)) {
            const b = stack.pop(), a = stack.pop();
            if (t === '+') stack.push(a + b);
            else if (t === '-') stack.push(a - b);
            else if (t === '*') stack.push(a * b);
            else stack.push(Math.trunc(a / b));
        } else stack.push(parseInt(t));
    }
    return stack[0];
};

File: scripts/test_solutions/150_evaluate_reverse_polish.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> stk;

        for (const string& t : tokens) {
            if (t == "+" || t == "-" || t == "*" || t == "/") {
                int b = stk.top(); stk.pop();
                int a = stk.top(); stk.pop();

                if (t == "+") stk.push(a + b);
                else if (t == "-") stk.push(a - b);
                else if (t == "*") stk.push(a * b);
                else stk.push(a / b);  // Integer division truncates toward zero in C++
            } else {
                stk.push(stoi(t));
            }
        }

        return stk.top();
    }
};
